ubccr/xdmod
# Make "My Profile" window stateless to avoid cutting off content.

This changes the "My Profile" window to be stateless (i.e., not to save its size and position for the next time it is opened) in order to fix a bug in which the content of the window was visually cut off.

## Motivation and Context


If the "My Profile" window was opened and the user switched to a tab that made the window smaller, e.g., the "Role Delegation" tab (when logged in as center director) or the "API Token" tab (introduced in https://github.com/ubccr/xdmod/pull/1694), and then the user dragged the window, closed it, and reopened it, the window would have the same size as when it was closed, which could cut off content in the "General" tab.

## Tests performed



1. Set up the test environment.
    1. New environment:
        1. Set up my port (9001) on `xdmod-dev` by running the script `update-open-xdmod-portal.php` with the branch `xdmod10.5-xsede`.
        1. Copy in the changed files from this Pull Request.
        1. Log in.
    1. Old environment: Spin up Docker containers from the Centos 7 image `tools-ext-01.ccr.xdmod.org/xdmod-10.0.0:centos7.9-0.6` and the Rocky 8 image `tools-ext-01.ccr.xdmod.org/xdmod10.0:rockylinux8.5-0.5`.
        1. Upgrade to the latest Open XDMoD 10.5 version.
        1. Merge in the changed files.
        1. Run the `~/bin/services start` command and open a browser to `localhost:8080`.
        1. Log in using center director credentials.
1. Click the "My Profile" button.
1. Click the "Role Delegation" or "API Token" tab.
1. Move the window.
1. Close the window.
1. Click the "My Profile" button.
1. Confirm the content in the "General" tab is not visually cut off.

When the window closes, it should still log out SSO users who do not have email addresses. To test this, in Docker containers (make sure to publish port `7000`) from the CentOS 7 image `tools-ext-01.ccr.xdmod.org/xdmod-10.0.0:centos7.9-0.6` and the Rocky 8 image `tools-ext-01.ccr.xdmod.org/xdmod10.0:rockylinux8.5-0.5`:
1. Edit the script `tests/ci/samlSetup.sh` to change the value of `email` from `saml.jackson@example.com` to an empty string and change all instances of `localhost` to the host port you are using for HTTPS, e.g., `localhost:8080`. If needed, also change the value of `DEFAULT_INSTALL_DIR`.
1. Run the script `tests/ci/samlSetup.sh`.
1. Browse to https://localhost:8080/.
1. Click "Sign In."
1. Click the "Login here" button.
1. Click "Sign in".
1. Confirm the "My Profile" window appears with the email address blank.
1. Click the "Close" button and confirm it warns you that you will be logged out.
1. Click "Yes" and confirm it logs you out.


## Checklist:
